### ISSUE TITLE
Add support for new trajectory formats

### ISSUE THREAD

**Issue by drroe** [Amber-MD/cpptraj#156]
Add support for the following trajectory formats:

- [x] TNG format
- [ ] MMTF format http://mmtf.rcsb.org/
- [x] Desmond DTR format

**Comment by rmcgibbo** [comment]
See also
- https://mailman-1.sys.kth.se/pipermail/gromacs.org_gmx-developers/2015-May/008300.html
- https://github.com/mdtraj/mdtraj/blob/master/mdtraj/formats/xtc/src

**Comment by drroe** [comment]
> https://mailman-1.sys.kth.se/pipermail/gromacs.org_gmx-developers/2015-May/008300.html

The `xdrfile` package seems like a good option, esp. since it's available with major Linux distros. It can be an optional package for people who want xtc support (and the trr support via `xdrfile` is probably better than what I currently have in cpptraj).

> https://github.com/mdtraj/mdtraj/blob/master/mdtraj/formats/xtc/src

These seem really cool - unfortunately I don't think the BSD license is compatible with GPL so cpptraj can't use these.

**Comment by mpharrigan** [comment]
I think GPL can use BSD, but not vice-versa

**Comment by rmcgibbo** [comment]
- xrd is the lower layer that handles getting floats/ints/bytes out of the file in an endian-independent way, but you still need to gromacs xtc code to decompress the coordinates.
- the 2-clause bsd is compatible with the gpl http://www.gnu.org/licenses/license-list.en.html#ModifiedBSD

**Comment by drroe** [comment]
> the 2-clause bsd is compatible with the gpl http://www.gnu.org/licenses/license-list.en.html#ModifiedBSD

Cool, good to know. Thanks for the info!

**Comment by drroe** [comment]
Note: this may be superseded by the TrajNG format eventually.

**Comment by hainm** [comment]
so should we close this issue? or change to TNG?

**Comment by mpharrigan** [comment]
xtc will be around for a while

**Comment by drroe** [comment]
> so should we close this issue? or change to TNG?

No. It's just to remind me. It will probably make sense to open a separate issue for TrajNG eventually, but it's probably still higher priority to get XTC supported at the moment.

**Comment by swails** [comment]
What is TrajNG?  Traj-Next-Gen?

**Comment by hainm** [comment]
Yeah

**Comment by drroe** [comment]
> What is TrajNG? Traj-Next-Gen?

I think that's exactly it. See: http://redmine.gromacs.org/issues/1926#note-6

**Comment by hainm** [comment]
Not sure why developers try to develop a bunch of file formats.

**Comment by hainm** [comment]
http://link.springer.com/article/10.1007%2Fs00894-010-0948-5

**Comment by swails** [comment]
I really hate when "next-gen" makes it into the name of a product.  It's an accurate descriptor for like 2 minutes.

**Comment by mabraham** [comment]
@hainm So they can have both good compression and good reading afterwards...

I'm happy to help get xtc and tng supported.

**Comment by drroe** [comment]
@mabraham That would be great! I've started playing around with using libxdrfile under the hood for this but I think it would be good to have someone who is more familiar with the formats take a look at this.

**Comment by drroe** [comment]
Changed title, XTC support added via #369.

**Comment by drroe** [comment]
TNG addressed by #768.

**Comment by drroe** [comment]
DTR added as of #796

**Comment by drroe** [comment]
MMTF is deprecated.
